From: Ximin Luo <infinity0@debian.org>
Date: Thu, 27 Oct 2016 16:04:19 +0200
Subject: Enable/disable the correct build commands

When building net.sf.j2s.ajax as a PDE project and then a J2S project
---
 Makefile                |  4 ++++
 switch-build-command.sh | 29 +++++++++++++++++++++++++++++
 2 files changed, 33 insertions(+)
 create mode 100644 switch-build-command.sh

diff --git a/Makefile b/Makefile
index 89f1fae..4ed4089 100644
--- a/Makefile
+++ b/Makefile
@@ -44,6 +44,8 @@ configure:
 	$(ECLIPSE_ANT) -f configure.xml $(CONFIGURE_FLAGS)
 
 build-plugins: configure
+	sh switch-build-command.sh sources/net.sf.j2s.ajax/.project \
+	  -net.sf.j2s.core.java2scriptbuilder +org.eclipse.jdt.core.javabuilder
 	set -e; for i in $(CORE_PLUGINS:%=sources/%); do \
 		( cd $$i && $(ECLIPSE_ANT_BUILD) $(CORE_FLAGS); ) \
 	done
@@ -64,6 +66,8 @@ local-install-plugins: build-plugins
 BADMETHOD1 = org/eclipse/jdt/internal/compiler/parser/TypeConverter.decodeType
 WORKAROUND1 = -vmargs -XX:CompileCommand=exclude,$(BADMETHOD1)
 build-libs: local-install-plugins
+	sh switch-build-command.sh sources/net.sf.j2s.ajax/.project \
+	  +net.sf.j2s.core.java2scriptbuilder -org.eclipse.jdt.core.javabuilder
 	test ! -f *err*.log
 	set -e; for i in $(CORE_J2SLIB); do \
 		$(ECLIPSE_J2S) -cmd build -path $$PWD/sources/$$i $(WORKAROUND1); \
diff --git a/switch-build-command.sh b/switch-build-command.sh
new file mode 100644
index 0000000..18b7ce6
--- /dev/null
+++ b/switch-build-command.sh
@@ -0,0 +1,29 @@
+#!/bin/sh
+set -e
+
+enable() {
+perl -0 -p \
+	-e 's|<!--\s*(<buildCommand>\s*<name>'"$1"'</name>.*?</buildCommand>\s*)\n\s*-->|\1|gs'
+}
+
+disable() {
+perl -0 -p \
+	-e 's|\n((\s*)<buildCommand>\s*<name>'"$1"'</name>.*?\n(\s*)</buildCommand>.*?\n)|\n\2<!--\n\1\3-->\n|sg' \
+| perl -0 -p \
+	-e 's|<!--\s*(<!--\s*<buildCommand>\s*<name>'"$1"'</name>.*?</buildCommand>\s*-->)\s*-->|\1|gs'
+}
+
+file="$1"
+shift
+
+while [ -n "$1" ]; do
+case "$1" in
++*) cmd=enable;name="${1#+}";;
+-*) cmd=disable;name="${1#-}";;
+*) false;;
+esac
+echo "$cmd" "$name"
+sed -e 's/\r$//g' "$file" | $cmd "$name" | sed -e 's/$/\r/g' > "$file.tmp"
+mv "$file.tmp" "$file"
+shift
+done
